llvm/lib/Transforms/InstCombine/InstCombineVectorOps.cpp

```diff
@@ -2900,6 +2900,17 @@ Instruction *InstCombinerImpl::visitShuffleVectorInst(ShuffleVectorInst &SVI) {
   if (Instruction *I = foldIdentityPaddedShuffles(SVI))
     return I;
 
+  if (Constant *C = dyn_cast<Constant>(RHS)) {
```

Comment:
```suggestion
  if (isa<Constant>(RHS)) {
```

Reply:
Good point! Will go with `match(RHS, M_Constant())` instead as that seems to match the style of the surrounding code pattern.